packages/flutter/lib/src/services/hardware_keyboard.dart

```diff
@@ -0,0 +1,894 @@
+///  * [KeyUpEvent], a key event representing the user
+///    releasing a key.
+///  * [HardwareKeyboard], which produces this event.
+class KeyDownEvent extends KeyEvent {
+  /// Creates a key event that represents the user pressing a key.
+  const KeyDownEvent({
+    required PhysicalKeyboardKey physicalKey,
+    required LogicalKeyboardKey logicalKey,
+    String? character,
+    required Duration timeStamp,
+    bool synthesized = false,
+  }) : super(
+         physicalKey: physicalKey,
+         logicalKey: logicalKey,
+         character: character,
+         timeStamp: timeStamp,
+         synthesized: synthesized,
+       );
+}
+
+/// An event indicating that the user has released a key on the keyboard.
+///
+/// See also:
+///
+///  * [KeyDownEvent], a key event representing the user
+///    pressing a key.
+///  * [KeyRepeatEvent], a key event representing the user
+///    holding a key, causing repeated events.
+///  * [HardwareKeyboard], which produces this event.
+class KeyUpEvent extends KeyEvent {
+  /// Creates a key event that represents the user pressing a key.
+  const KeyUpEvent({
+    required PhysicalKeyboardKey physicalKey,
+    required LogicalKeyboardKey logicalKey,
+    required Duration timeStamp,
+    bool synthesized = false,
+  }) : super(
+         physicalKey: physicalKey,
+         logicalKey: logicalKey,
+         timeStamp: timeStamp,
+         synthesized: synthesized,
+       );
+}
+
+/// An event indicating that the user has been holding a key on the keyboard
+/// and causing repeated events.
+///
+/// See also:
+///
+///  * [KeyDownEvent], a key event representing the user
+///    pressing a key.
+///  * [KeyUpEvent], a key event representing the user
+///    releasing a key.
+///  * [HardwareKeyboard], which produces this event.
+class KeyRepeatEvent extends KeyEvent {
+  /// Creates a key event that represents the user pressing a key.
+  const KeyRepeatEvent({
+    required PhysicalKeyboardKey physicalKey,
+    required LogicalKeyboardKey logicalKey,
+    String? character,
+    required Duration timeStamp,
+  }) : super(
+         physicalKey: physicalKey,
+         logicalKey: logicalKey,
+         character: character,
+         timeStamp: timeStamp,
+       );
+}
+
+/// The signature for [HardwareKeyboard.addHandler], a callback to to decide whether
+/// the entire framework handles a key event.
+typedef KeyEventCallback = bool Function(KeyEvent event);
+
+/// Manages key events from hardware keyboards.
+///
+/// [HardwareKeyboard] manages all key events of the Flutter application from
+/// hardware keyboards (in contrast to on-screen keyboards). It receives key
+/// data from the native platform, dispatches key events to registered
+/// handlers, and records the keyboard state.
+///
+/// To stay notified whenever keys are pressed, held, or released, add a
+/// handler with [addHandler]. It is recommended to use [Focus] instead to be
+/// notified only when specific part of the app is focused.
```

Comment:
```suggestion
/// handler with [addHandler]. To only be notified when a specific part of the app
/// is focused, use a [Focus] widget's `onFocusChanged` attribute instead of [addHandler].
/// Handlers should be removed with [removeHandler] when notification is no longer 
/// necessary, or when the handler is being disposed.
```